docs/source/integrations/middleware.md

```diff
@@ -395,7 +395,7 @@ async function startApolloServer(typeDefs, resolvers) {
   const app = new Koa();
   server.applyMiddleware({ app });
   httpServer.on('request', app.callback());
-  await new Promise(resolve => httpServer.listen({ port: 4000 }, resolve));
+  await new Promise(resolve => httpServer.listen({ port: 4000 }, () => resolve(undefined)));
```

Comment:
FWIW, in TyepScript you can also do `await new Promise<void>` for this, right? That's what I tend to do.

This example code isn't TS though so that's not legal. (We should rewrite all our docs to use TS but we aren't.)

So that makes this a bit odd — the current code isn't legal TS, but your suggestion wouldn't be idiomatic TS, but the idiomatic TS would not be legal JS.  Thoughts?

(In any case, there are two other instances in this page so we should make them all consistent.)

Reply:
I think we've generally fixed this in the docs via `await new Promise<void>`